### PR TITLE
[Bart] Fix: put dummy_inputs on correct device

### LINKED ISSUE #3381: [BART] test_dummy_inputs fails on GPU
```
RuntimeError: Expected object of device type cuda but got device type cpu for argument #3 'index' in call to _th_index_select

```
Easy fix, but putting here so I don't forget!

### PR DESCRIPTION
**Pull request by sshleifer**
This fixes `test_dummy_inputs`, which was failing on GPU because dummy inputs were put on CPU even if model was on GPU.

### CODE REVIEW

File: src/transformers/modeling_bart.py [hunk 1]
```diff
@@ -138,6 +138,10 @@ def dummy_inputs(self):
         }
         return dummy_inputs
 
+    @property
```
Comment (sshleifer) [inline src/transformers/modeling_bart.py]: would be happy to move this to `PretrainedModel`

Comment (sshleifer) [reply src/transformers/modeling_bart.py]: This and `next(self.parameters()).dtype` have roughly 20 appearance in the repo.

Comment (thomwolf) [reply src/transformers/modeling_bart.py]: yes, you can move this to `PretrainedModel` indeed

**Comment by sshleifer** [comment]
Merging with deep suspicion that circleci failure is spurious.

**Pull request merged**